staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -72,6 +74,7 @@ func NewAllocator(ctx context.Context,
 	return &Allocator{
 		adminAccessEnabled:     adminAccessEnabled,
 		prioritizedListEnabled: prioritizedListEnabled,
+		deviceBindingEnabled:   deviceBindingEnabled,
```

Comment:
There is one other thing that the allocator must do: if DRABindingConditions **or DRAResourceClaimDeviceStatus** are disabled, devices depending on binding conditions cannot be allocated because the scheduler is not supposed to do binding conditions (first feature gate) or the binding conditions in the status wouldn't be available (second feature gate).

This can occur during downgrade or version skew scenarios.

Reply:
I add the check for DRAResourceClaimDeviceStatus.